### DISCUSSION TITLE
Why async entrypoints if code is sync

### DISCUSSION THREAD

**Discussion by jalvespinto** [pgorecki/python-ddd#17] [Q&A]
I am developing an app on fastapi and basically following the design in [architecture patterns with python](https://www.cosmicpython.com/book/preface.html). Now I will try to make it async and during my study I found this repo. I have little experience with async, but if I understood right your code is thread safe, but sync and yet you are using async entrypoints. Wouldn't this by a issue when there is IO events since requests to a endpoint would run in the same thread?

**Comment by pgorecki** [comment]
Ideally, command and query execution should be async, but this is not my priority at the moment :) I will look into it in the future, when all core functionalities are ready.
